src/xpub/repo.rs

```diff
@@ -17,71 +28,45 @@ impl XPubs {
     }
 
     #[instrument(name = "xpubs.persist", skip(self))]
-    pub async fn persist(&self, xpub: NewAccountXPub) -> Result<XPubId, XPubError> {
-        let mut tx = self.pool.begin().await?;
-        let ret = self.persist_in_tx(&mut tx, xpub).await?;
-        tx.commit().await?;
-        Ok(ret)
+    pub async fn persist(&self, xpub: NewXpub) -> Result<XPubId, XpubError> {
+        let xpub = self.create(xpub).await?;
```

Comment:
do we even need this fn at all ? 
if we need to call create why not just use repo.create() ?